include/verilated_timing.h

```diff
@@ -187,15 +198,25 @@ class VlDelayScheduler final {
             VlProcessRef process;  // Data of the suspended process, null if not needed
             VlDelayedCoroutineQueue& queue;
             uint64_t delay;
+            VlDelayPhase phase;
             VlFileLineDebug fileline;
 
             bool await_ready() const { return false; }  // Always suspend
             void await_suspend(std::coroutine_handle<> coro) {
-                queue.emplace(delay, VlCoroutineHandle{coro, process, fileline});
+                queue.emplace(VlDelayTime{delay, phase},
+                              VlCoroutineHandle{coro, process, fileline});
             }
             void await_resume() const {}
         };
-        return Awaitable{process, m_queue, m_context.time() + delay,
+
+        VlDelayPhase phase = (delay == 0) ? VlDelayPhase::INACTIVE : VlDelayPhase::ACTIVE;
+        if (phase == VlDelayPhase::INACTIVE) {
+            VL_WARN_MT(filename, lineno, VL_UNKNOWN,
+                       "Ran into #0 delay. However, process "
+                       "will be resumed before combinational logic evaluation.");
```

Comment:
Is this something that's the user's fault, or internal?  Can we suggest some repairing action or what user is to do?